proposals/constant_interpolated_strings.md

```diff
@@ -0,0 +1,121 @@
+# Constant Interpolated Strings
+
+* [x] Proposed
+* [ ] Prototype: [Not Started](https://github.com/kevinsun-dev/roslyn/BRANCH_NAME)
+* [ ] Implementation: [Not Started](https://github.com/dotnet/roslyn/BRANCH_NAME)
+* [ ] Specification: [Not Started](pr/1)
+
+## Summary
+[summary]: #summary
+
+Enables constants to be generated from interpolated strings of type string constant.
+
+## Motivation
+[motivation]: #motivation
+
+The following code is already legal:
+```
+public class C
+{
+    const string S1 = "Hello world";
+    const string S2 = "Hello" + " " + "World";
+    const string S3 = S1 + " Kevin, welcome to the team!";
+}
+```
+However, there have been many community requests to make the following also legal:
+```
+public class C
+{
+    const string S1 = $"Hello world";
+    const string S2 = $"Hello{" "}World";
+    const string S3 = $"{S1} Kevin, welcome to the team!";
+}
+```
+This proposal represents the next logical step for constant string generation, where existing string syntax that works in other situations is made to work for constants.
+
+## Detailed design
+[design]: #detailed-design
+
+The following represent the updated specifications for constant expressions under this new proposal.
+
+### Constant Expressions
+
+A *constant_expression* is an expression that can be fully evaluated at compile-time.
+
+```antlr
+constant_expression
+    : expression
+    ;
+```
+
+A constant expression must be the `null` literal or a value with one of  the following types: `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `char`, `float`, `double`, `decimal`, `bool`, `object`, `string`, or any enumeration type. Only the following constructs are permitted in constant expressions:
```

Comment:
Looks like this is coming mostly from the C# spec, should attribute that in the writing here. Basically talk about which section of which spec version you got this from.

Reply:
Added direct link to section of current spec.